st2actions/st2actions/scheduler/queue_handler.py

```diff
@@ -23,42 +23,42 @@
 from st2common.services import action as action_service
 from st2common.services import policies as policy_service
 from st2common.persistence.liveaction import LiveAction
-from st2common.transport import consumers
-from st2common.transport import utils as transport_utils
+from st2common.util import execution_queue_db as exdb
 from st2common.util import action_db as action_utils
-from st2common.transport.queues import ACTIONSCHEDULER_REQUEST_QUEUE
+from st2common.services.coordination import LockAcquireError
 
 __all__ = [
-    'ActionExecutionScheduler',
-    'get_scheduler'
+    'ExecutionQueueHandler',
+    'get_handler'
 ]
 
 
 LOG = logging.getLogger(__name__)
 
 
-class ActionExecutionScheduler(consumers.MessageHandler):
-    message_type = LiveActionDB
-
-    def process(self, request):
-        """Schedules the LiveAction and publishes the request
-        to the appropriate action runner(s).
-
-        LiveAction in statuses other than "requested" are ignored.
-
-        :param request: Action execution request.
-        :type request: ``st2common.models.db.liveaction.LiveActionDB``
-        """
-
-        if request.status != action_constants.LIVEACTION_STATUS_REQUESTED:
-            LOG.info('%s is ignoring %s (id=%s) with "%s" status.',
-                     self.__class__.__name__, type(request), request.id, request.status)
-            return
-
+class ExecutionQueueHandler(object):
+    def __init__(self):
+        self.message_type = LiveActionDB
+        self._shutdown = False
+
+    def loop(self):
+        LOG.debug('Entering scheduler loop')
+        while self._shutdown is not True:
+            # TODO: Make config setting
+            eventlet.greenthread.sleep(0.25)
+            try:
+                execution = exdb.pop_next_execution()
+                if execution:
+                    eventlet.spawn(self._handle_execution, execution)
```

Comment:
It looks like you already call `loop` in a greenthread so there is probably no need to also use it here.